### ISSUE TITLE
Change Base Model Type to something more general

### ISSUE THREAD

**Issue by laxitshahi** [llm-proxy/llm-proxy#64]
Because the llm proxy may later use different types of models, we want the base ABC to be renamed into something more general. For example, BaseLLM, BaseInference, Base...

**Note**: This should likely be done after the code from the different branches have been merged into the main branch

**Comment by laxitshahi** [comment]
Depends on #30 #32 (merged)
